jfrog-oauth/main.tf

```diff
@@ -0,0 +1,77 @@
+terraform {
+  required_version = ">= 1.0"
+
+  required_providers {
+    coder = {
+      source  = "coder/coder"
+      version = ">= 0.12"
+    }
+  }
+}
+
+variable "jfrog_url" {
+  type        = string
+  description = "JFrog instance URL. e.g. https://jfrog.example.com"
+}
+
+variable "username_field" {
+  type        = string
+  description = "The field to use for the artifactory username. i.e. Coder username or email."
+  default     = "username"
+  validation {
+    condition     = can(regex("^(email|username)$", var.username_field))
+    error_message = "username_field must be either 'email' or 'username'"
+  }
+}
```

Comment:
Why is this necessary? Can we read the username from the token if its a JWT? [example](https://github.com/coder/customers/issues/248#issuecomment-1679258824)

Perhaps Coder should expose the userinfo/token as a data source

Reply:
In our SaaS instance we have usernames set to emails but in the self hosted when we sign up using the GitHub or Google OAuth the usernames are equal to usernames. 

Our `coder_external_auth` data doesn't expose the username and only exposes `access_token`.

Reply:
If you put the token into http://jwt.io, does it include the user's email?